llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp

```diff
@@ -1991,7 +1993,7 @@ static bool annotateAllFunctions(
   M.setProfileSummary(PGOReader->getSummary(IsCS).getMD(M.getContext()),
                       IsCS ? ProfileSummary::PSK_CSInstr
                            : ProfileSummary::PSK_Instr);
-  PSI->refresh();
+  auto *PSI = AbandonAndRefreshPSI(M);
```

Comment:
Is this change equivalent? The refresh API early returns if profile summary exists.

Reply:
It's not, but there shouldn't have been a profile summary at this point to begin with. I think the PR expresses the intent.

Reply:
(offline discussion) adding an assert that we had no PSI.

Reply:
actually there could be one when we do csfdo, for example. So we do mean "abandon"